node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala

```diff
@@ -3,34 +3,48 @@ package coop.rchain.node.web
 import cats.effect.Sync
 import cats.syntax.all._
 import cats.~>
-import io.circe.generic.semiauto._
 import com.google.protobuf.ByteString
 import coop.rchain.casper.PrettyPrinter
 import coop.rchain.node.api.WebApi
 import coop.rchain.node.api.WebApi._
+import coop.rchain.shared.Log
+import io.circe.generic.semiauto._
 import org.http4s.{HttpRoutes, Response}
+
 object WebApiRoutes {
-  import WebApiSyntax._
 
-  def service[F[_]: Sync, M[_]: Sync](
+  def service[F[_]: Sync: Log, M[_]: Sync](
       webApi: WebApi[M]
   )(implicit mf: M ~> F): HttpRoutes[F] = {
+    import coop.rchain.casper.protocol.{BlockInfo, LightBlockInfo}
+    import io.circe._
     import io.circe.generic.auto._
     import io.circe.syntax._
-    import io.circe._
     import org.http4s.circe._
-    import org.http4s.{
-      EntityDecoder,
-      EntityEncoder,
-      InvalidMessageBodyFailure,
-      QueryParamDecoder,
-      Request
-    }
-    import coop.rchain.casper.protocol.{BlockInfo, BondInfo, DeployInfo, LightBlockInfo}
+    import org.http4s.{EntityDecoder, EntityEncoder, InvalidMessageBodyFailure, Request}
 
     val dsl = org.http4s.dsl.Http4sDsl[F]
     import dsl._
 
+    implicit class ResponseErrorHandler(val res: F[Response[F]]) {
+      def getErrorMsg(ex: Throwable) = if (ex.getMessage == null) ex.toString else ex.getMessage
+      def handleResponseError =
+        res
+          .onError {
+            case ex: Throwable => Log[F].error("HTTP API response error", ex)
```

Comment:
Logging all response errors is not such a good idea. Every call to _findDeploy_, when deploy is not proposed, will be logged as error.

Reply:
I changed this to log only unanticipated errors.